### ISSUE TITLE
Is there a size limit for blocks?

### ISSUE THREAD

**Issue by usd-yamazaki** [openethereum/openethereum#11334]
- **Parity Ethereum version**: 2.5.10
- **Operating system**: Linux
- **Installation**: one-line installer
- **Fully synchronized**: yes
- **Network**: private PoA(AuRa)
- **Restarted**: yes

One Authority and one Fullnode (not sealing) are used.
I sent 1 million transactions with sendRawTransaction to measure the TPS of the contract.
All transactions were successfully sealed and Authority continued to work normally, but Fullnode synchronization stopped.
If you check blocks on the Authority that cannot be synchronized, one block contains 220,000 transactions.

Transactions have a 300KB size limit by default, but is there a block size limit as well?
Is there a way to change it?

**Comment by dvdplm** [comment]
[Yes](https://ethgasstation.info/blog/ethereum-block-size/).

**Comment by usd-yamazaki** [comment]
> [はい](https://ethgasstation.info/blog/ethereum-block-size/)。

Thank you, but there was no answer here.
I am successful in block seal. This means that you are not restricted by gas.
By the way, block gas limit is set to 9007199254740991.

**Comment by usd-yamazaki** [comment]
There is no particular error in the log on the Fullnode side, but I found that the following log frequently occurs in the log on the Authority side.

```
2019-12-16 14:42:25  IO Worker #1 DEBUG sync  "Error sending block bodies: Error(OversizedPacket, State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })"
2019-12-16 14:42:44  IO Worker #0 DEBUG sync  "Error sending block bodies: Error(OversizedPacket, State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })"
2019-12-16 14:43:04  IO Worker #3 DEBUG sync  "Error sending block bodies: Error(OversizedPacket, State { next_error: None, backtrace: InternalBacktrace { backtrace: None } })"
```

**Comment by usd-yamazaki** [comment]
I found that block synchronization is limited to 24bit (16MB) size by MAX_PAYLOAD_SIZE, which is defined in RLPx.
https://github.com/paritytech/parity-ethereum/blob/f1f64930cfcd5311bc5dece78321efbd683013d4/util/network-devp2p/src/connection.rs#L43
https://github.com/ethereum/devp2p/blob/master/rlpx.md#message-encoding

I added logs to the source and checked the size of the block I was trying to synchronize and it was over 100MB.
I think that one of the following is necessary so that the block which parity created can be synchronized normally.


- Change MAX_PAYLOAD_SIZE with chain spec json etc.
- Make sure that blocks exceeding MAX_PAYLOAD_SIZE can be sent separately.
- Prevent blocks that exceed MAX_PAYLOAD_SIZE from being sealed.
